src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -295,6 +328,94 @@ void ScheduleTask(DateTimeOffset utcNow)
             }
         }
 
+        private (long, long) Sum()
+        {
+            lock (_allStats!)
+            {
+                long hits = _accumulatedStats!.Hits;
+                long misses = _accumulatedStats.Misses;
+
+                foreach (WeakReference<Stats> wr in _allStats)
+                {
+                    if (wr.TryGetTarget(out Stats? stats))
+                    {
+                        hits += Interlocked.Read(ref stats.Hits);
+                        misses += Interlocked.Read(ref stats.Misses);
+                    }
+                }
+
+                return (hits, misses);
+            }
+        }
+
+        private void Hit()
```

Comment:
This method is on the hot path, it would be nice if it was getting inlined when stats are disabled. @maryamariyan can you see it in the profiles when stats are disabled?

If it does not get inlined we could consider forcing it to inline by using the attribute.